### ISSUE TITLE
[Enhancement]: FSE Themes: add Prevent Removal lock for the Post Content Block in Single and Page templates

### ISSUE THREAD

**Issue by liviopv** [Automattic/themes#7073]
## What

We should implement a "Prevent Removal" block lock for the Post Content Block in all FSE themes developed by Automattic.

## Why

Templates and Template Blocks are complex concepts for novice customers and old-school bloggers (a sizeable part of WordPress.com customer base), and it's not uncommon for users to find themselves removing the Post Content Block from Page and Single. This causes frustrating situations where the content user has added to Posts and Pages cannot be seen on the front end.

A sample of customer interactions about this issue can be found in the `fse_template_vs_post_content` Happychat tag, but many more likely don't have this tag.

While a better Template/Content editing experience should (and will) come from core, WordPress.com can still provide a better experience to our customers by offering FSE themes with added friction to removing this block.

Some themes already implement locks for different elements, so this proposal has precedent in theme development. Kerr, for instance, includes a lock in the Group Block that wraps the Post Content in the `page` template. Iotix has a lock in the `single` template for Group Block that wraps the Post Title, Post Meta and Featured Image, but not for the Post Content.

## How

- [ ] Adjust all a8c FSE theme annotations (is that the correct term?) and add a "Prevent Removal" lock placed in the Post Content Block on the `single` and `page` templates
- [ ] Document this as a guideline for theme development at a8c; raise awareness among theme designers for in-development themes.

## Related
https://github.com/WordPress/gutenberg/issues/40618

**Comment by cuemarie** [comment]
📌 ACTIONS
- Feature request kept

**Comment by annezazu** [comment]
Wanted to note that currently there's a warning when this block is attempted to be removed to add friction: https://github.com/WordPress/gutenberg/pull/51145 I'm curious if this is helping this situation. For example, I'm keen to update the copy and am open to feedback there: https://github.com/WordPress/gutenberg/issues/52392

**Comment by cuemarie** [comment]
@liviopv from the looks of it, it seems like https://github.com/WordPress/gutenberg/pull/51145 does address this need. I'm of the mind that this enhancement request is addressed, and then you or other folks might want to chime in on the discussion Anne mentioned on the copy of the warning - https://github.com/WordPress/gutenberg/issues/52392.  What do you think?

**Comment by liviopv** [comment]
@cuemarie yes, it addresses it. We'd only need to keep this one if we determine it is still a big problem and we want to double-down on adding friction to removing these blocks. cc @retnonindya as this touches on theme experience as well.
